src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/AzureWorkload/ItemTests.ps1

```diff
@@ -0,0 +1,487 @@
+			-VaultId $vault.ID `
+			-Container $container `
+			-WorkloadType "MSSQL";
+		Assert-True { $newprotectableItems.count -ge $protectableItems.count }
+	}
+	finally
+	{
+		#Unregister container
+		Unregister-AzRecoveryServicesBackupContainer `
+			-VaultId $vault.ID `
+			-Container $container
+	}
+}
+
+function Test-AzureVmWorkloadEnableProtectableItem
+{
+	try
+	{
+		$vault = Get-AzRecoveryServicesVault -ResourceGroupName $resourceGroupName -Name $vaultName
+		
+		$policy = Get-AzRecoveryServicesBackupProtectionPolicy `
+			-VaultId $vault.ID `
+			-Name $policyName
+		
+		$container = Register-AzRecoveryServicesBackupContainer `
+			-ResourceId $resourceId `
+			-BackupManagementType AzureWorkload `
+			-WorkloadType MSSQL `
+			-VaultId $vault.ID
+		Assert-AreEqual $container.Status "Registered";
+
+		$protectableItems = Get-AzRecoveryServicesBackupProtectableItem `
+			-VaultId $vault.ID `
+			-Container $container `
+			-WorkloadType "MSSQL";
+
+		Enable-AzRecoveryServicesBackupProtection `
+			-VaultId $vault.ID `
+			-Policy $policy `
+			-ProtectableItem $protectableItems[1]
+
+		$item = Get-AzRecoveryServicesBackupItem `
+			-VaultId $vault.ID `
+			-Container $container `
+			-WorkloadType MSSQL;
+		Assert-True { $item.Name -contains $protectableItems[1].Name }
+		Assert-True { $item.LastBackupStatus -eq "IRPending" }
+		Assert-True { $item.ProtectionPolicyName -eq $policyName }
+	}
+	finally
+	{
+		Cleanup-Vault $vault $item $container
+	}
+}
+
+function Test-AzureVmWorkloadEnableAutoProtectableItem
+{
+	try
+	{
+		$vault = Get-AzRecoveryServicesVault -ResourceGroupName $resourceGroupName -Name $vaultName
+		
+		$policy = Get-AzRecoveryServicesBackupProtectionPolicy `
+			-VaultId $vault.ID `
+			-Name $policyName
+		
+		$container = Register-AzRecoveryServicesBackupContainer `
+			-ResourceId $resourceId `
+			-BackupManagementType AzureWorkload `
+			-WorkloadType MSSQL `
+			-VaultId $vault.ID
+		Assert-AreEqual $container.Status "Registered";
+
+		$protectableInstances = Get-AzRecoveryServicesBackupProtectableItem `
+			-VaultId $vault.ID `
+			-Container $container `
+			-WorkloadType "MSSQL" `
+			-ItemType "SQLInstance";
+		
+		$protectableItems = Get-AzRecoveryServicesBackupProtectableItem `
+			-VaultId $vault.ID `
+			-Container $container `
+			-WorkloadType "MSSQL";
+
+		Enable-AzRecoveryServicesBackupProtection `
+			-VaultId $vault.ID `
+			-Policy $policy `
+			-ProtectableItem $protectableItems[1]
+
+		$item = Get-AzRecoveryServicesBackupItem `
+			-VaultId $vault.ID `
+			-Container $container `
+			-WorkloadType MSSQL;
+
+		Get-AzRecoveryServicesBackupProtectableItem `
+			-VaultId $vault.ID `
+			-Container $container `
+			-WorkloadType "MSSQL" `
+			-ItemType "SQLInstance" | Enable-AzRecoveryServicesBackupAutoProtection `
+				-VaultId $vault.ID `
+				-BackupManagementType "AzureWorkload" `
+				-WorkloadType "MSSQL" `
+				-Policy $policy;
+
```

Comment:
What is your approach to check the cmdlet works as expected without using `Assert-*` through the test?

Reply:
AutoProtection APIs are intents rather than long-running operations and we dont have any cmdlet to get intent's status. We have added this in our ToDos to come up with a better test plan.